### PR TITLE
CSHARP-2930: Change default GuidRepresentationMode to V3 and default GuidRepresentation to Unspecified.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

**Review by sanych-sun — COMMENTED** [review]
Please check cake script, there are something that seems to be related to Guid representation.

**Comment by rstam** [comment]
> Please check cake script, there are something that seems to be related to Guid representation.

I removed unused Guid related parts of build.cake.

**Review by sanych-sun — APPROVED** [review]
LGTM

File: tests/MongoDB.Bson.Tests/Serialization/Serializers/GuidSerializerTests.cs [hunk 1]
```diff
@@ -98,80 +98,31 @@ public void Representation_should_return_expected_result(
 
         public static IEnumerable<object[]> Deserialize_should_return_expected_result_when_representation_is_binary_MemberData()
         {
-            var data = new TheoryData<GuidRepresentationMode, GuidRepresentation, GuidRepresentation, GuidRepresentation, GuidRepresentation>();
+            var data = new TheoryData<GuidRepresentation>();
```
Comment (BorisDog) [inline tests/MongoDB.Bson.Tests/Serialization/Serializers/GuidSerializerTests.cs:101]: For your consideration: optional shorter form
`Deserialize_should_return_expected_result_when_representation_is_binary_MemberData() =>
EnumHelper.GetValues<GuidRepresentation>().Except([GuidRepresentation.Unspecified]).Select(v => new object[] { v });`

or
`[Values(GuidRepresentation.CSharpLegacy, GuidRepresentation.JavaLegacy, GuidRepresentation.PythonLegacy, GuidRepresentation.Standard)]`

File: tests/MongoDB.Bson.Tests/Serialization/Serializers/ObjectSerializerTests.cs [hunk 2]
```diff
@@ -701,15 +647,11 @@ public void Deserialize_binary_data_should_return_expected_result_when_guid_repr
                 var expectedResult = GuidConverter.FromBytes(guidBytes, guidRepresentation);
                 result.Should().Be(expectedResult);
             }
-#pragma warning restore 618
         }
 
         [Fact]
-        [ResetGuidModeAfterTest]
         public void Deserialize_binary_data_should_throw_when_guid_representation_is_unspecified_and_mode_is_v3()
```
Comment (BorisDog) [inline tests/MongoDB.Bson.Tests/Serialization/Serializers/ObjectSerializerTests.cs:653]: just `Deserialize_binary_data_should_throw_when_guid_representation_is_unspecified` ?

File: tests/MongoDB.Bson.Tests/Serialization/Serializers/ObjectSerializerTests.cs [hunk 3]
```diff
@@ -761,75 +693,19 @@ public void Deserialize_binary_data_should_throw_when_guid_representation_is_spe
...
-        }
-
-        [Theory]
-        [ParameterAttributeData]
-        [ResetGuidModeAfterTest]
         public void Serialize_guid_should_have_expected_result_when_guid_representation_is_specified(
-            [ClassValues(typeof(GuidModeValues))]
-            GuidMode mode,
             [Values(GuidRepresentation.CSharpLegacy, GuidRepresentation.JavaLegacy, GuidRepresentation.PythonLegacy, GuidRepresentation.Standard, GuidRepresentation.Unspecified)]
             GuidRepresentation writerGuidRepresentation,
```
Comment (BorisDog) [inline tests/MongoDB.Bson.Tests/Serialization/Serializers/ObjectSerializerTests.cs:702]: `writerGuidRepresentation` not used anymore?

File: tests/MongoDB.Bson.Tests/Serialization/Serializers/ObjectSerializerTests.cs [hunk 4]
```diff
@@ -724,32 +666,22 @@ public void Deserialize_binary_data_should_throw_when_guid_representation_is_uns
...
         }
 
         [Theory]
         [ParameterAttributeData]
-        [ResetGuidModeAfterTest]
         public void Deserialize_binary_data_should_throw_when_guid_representation_is_specified_and_sub_type_is_not_expected_sub_type(
-            [ClassValues(typeof(GuidModeValues))]
-            GuidMode mode,
             [Values(GuidRepresentation.CSharpLegacy, GuidRepresentation.JavaLegacy, GuidRepresentation.PythonLegacy, GuidRepresentation.Standard)]
             GuidRepresentation readerGuidRepresentation,
```
Comment (BorisDog) [inline tests/MongoDB.Bson.Tests/Serialization/Serializers/ObjectSerializerTests.cs:675]: `readerGuidRepresentation` not used anymore?

**Review by BorisDog — CHANGES_REQUESTED** [review]

**Review by BorisDog — APPROVED** [review]
LGTM.

**Pull request merged**